prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -757,7 +757,14 @@ void rxr_pkt_handle_send_error(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entr
 	if (RXR_GET_X_ENTRY_TYPE(pkt_entry) == RXR_TX_ENTRY) {
 		tx_entry = pkt_entry->x_entry;
 		if (prov_errno == IBV_WC_RNR_RETRY_EXC_ERR &&
-		    ep->handle_resource_management == FI_RM_ENABLED) {
```

Comment:
This commit correctly handle the error code `IBV_WC_RNR_RETRY_EXC_ERR` correctly, but other type of rdma-core error as still not handled correctly.

I suggest that we take this opportunity to create a ibv error to fi_errno conversion table to correctly handle all ibv_wc error code.

Reply:
Which ones do you have in mind? I'm not sure if there are other rdma-core errors we want to return directly to the app. Maybe invalid key or registration?

Reply:
Any rdma-core error we encountered during, would cause an error CQ entry to be written.

From rdma-core code, the following are the possible errors:

                IBV_WC_WR_FLUSH_ERR;
                IBV_WC_LOC_QP_OP_ERR;
                IBV_WC_LOC_PROT_ERR;
                IBV_WC_LOC_LEN_ERR;
                IBV_WC_REM_ABORT_ERR;
                IBV_WC_RNR_RETRY_EXC_ERR;
                IBV_WC_REM_INV_RD_REQ_ERR;
                IBV_WC_BAD_RESP_ERR;
                IBV_WC_REM_INV_REQ_ERR;
                IBV_WC_GENERAL_ERR;

I think most common ones are BV_WC_RNR_RETRY_EXC_ERR and IBV_WC_REM_INV_RD_REQ_ERR (which we encountered when destine QP has been destroyed) 

Other errors are less common, but certainly possible to happen, especially when during development. For example the IBV_WC_LOC_QP_OP_ERR and IBV_WC_LOC_PROT_ERR will be encounter is wrong AHN or Local key is passed.

Reply:
A lot of those are unrecoverable and should just map to FI_EIO. Of course, we should pass along any additional information we can via prov_errno for debugging. I'd like to write another patch to do that for rdma-core and efa provider specific errors.

I'm not sure what an application will do with RD_REQ_ERR as that also seems unrecoverable.

Reply:
> A lot of those are unrecoverable and should just map to FI_EIO. 

Indeed, though I wonder if there are value to assign unique error code to them, and maybe have strerror print unique error message for them. This would help future debug IMO.

> I'm not sure what an application will do with RD_REQ_ERR as that also seems unrecoverable.

Indeed the error is unrecoverable. However, currently a user get the error entry with `errno` being 15, which is an undefined error code for libfabric. So, having a different error code would help debug.

Reply:
> Indeed, though I wonder if there are value to assign unique error code to them, and maybe have strerror print unique error message for them. This would help future debug IMO.

Yep, we can do that. Return a more generic error code (like FI_EIO) since there's nothing the application can do to recover, but provide extra info on what specifically happened in the provider for the error message.

> Indeed the error is unrecoverable. However, currently a user get the error entry with errno being 15, which is an undefined error code for libfabric. So, having a different error code would help debug.

Ah, I understand now. Sounds like a bug that I can go track down.